tests/test_datacollector.py

```diff
@@ -51,11 +51,21 @@ def __init__(self):
             {
                 "total_agents": lambda m: m.schedule.get_agent_count(),
                 "model_value": "model_val",
+                "model_calc": self.schedule.get_agent_count(),
```

Comment:
Yes but this is to test the different ways users can pass in objects to make sure they are all working. In this case they are different paths to the same output, but in different models this will not always be the case.

Reply:
But here you are calling the function at init time - Even if in another model the number of agents changes "model_calc" will be fixed to 10.

Reply:
Right, I will have to look at how it is called and make it cleaner

Reply:
I explored this and the datacollector.collect() placed in the step function does call the time.get_agent_count() at each step

Reply:
This must be from the `lambda m: m.schedule.get_agent_count()` above. 
At runtime you literally pass `{"model_calc": 10}` to the datacollector, so it has no way of knowing which function to call.

Reply:
I am not sure how I kept messing this up. However, to rectify I have now wrapped this data_collector variation in a decorator so it will call the function and updated the test_datacollector so it tests for the removal of an agent.